jaus_ros_bridge/src/SetFinDeflection.cpp

```diff
@@ -60,37 +62,36 @@ void SetFinDeflection::ProcessData(char* message) {
   int index = 6;                             // read data starts from index 6
   if (_PresenceVector != nullptr) delete _PresenceVector;
   int16_t pv = buildPresenceVector(message, index);
-  // printf("Presence vector is: %d \n", pv);
-
-  _finId = (int8_t)message[index++];  // 1 byte
-  printf("FinID is: %d \n", _finId);
 
+  _finId = (int8_t)message[index++];            // 1 byte
   _deflectionAngle = (int8_t)message[index++];  // 1 byte
-  printf("_deflectionAngle in degree is: %d \n", _deflectionAngle);
 
-  fin_control::SetAngle msg;
-  msg.ID = _finId;
-  msg.angle_in_radians = JausDataManager::degreesToRadians(_deflectionAngle);
-  // int count = 0;
-  // while(count<5)
-  //{
-  _publisher_setAngle.publish(msg);
-  //    count++;
+  _finAngles.push_back(JausDataManager::degreesToRadians(_deflectionAngle));
 
-  //    sleep(0.05);
-  //    printf("sleeping for 50ms. count= %d \n", count);
-  //}
+  // Check that all angles were set
+  if (_finAngles.size() == 4)
```

Comment:
@patapufate hmm, shouldn't this be setting the corresponding angle only?

Reply:
Answering myself, no, it should not. `SetFinDeflection` handles all four (4).

Let's just add a check that ensures that the incoming fin ID is the one we expect it to be. If it changes, we'll know.

Reply:
Added assertion in bc1e7f0